core/src/test/java/org/kiwix/kiwixmobile/core/dao/HistoryDaoTest.kt

```diff
@@ -0,0 +1,171 @@
+package org.kiwix.kiwixmobile.core.dao
+
+import io.mockk.every
+import io.mockk.mockk
+import io.objectbox.BoxStore
+import io.objectbox.kotlin.boxFor
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.kiwix.kiwixmobile.core.dao.entities.MyObjectBox
+import org.kiwix.kiwixmobile.core.history.HistoryListItem
+import org.kiwix.kiwixmobile.core.history.HistoryListItem.HistoryItem
+import java.io.File
+
+internal class HistoryDaoTest {
+  private val testDirectory = File("src/test/java/org/kiwix/kiwixmobile/core/dao/test-db")
+  private lateinit var boxStore: BoxStore
+  private lateinit var historyDao: HistoryDao
+
+  @BeforeEach
+  fun setUp() {
+    BoxStore.deleteAllFiles(testDirectory)
+    boxStore = MyObjectBox.builder().directory(testDirectory).build()
+    historyDao = HistoryDao(boxStore.boxFor())
```

Comment:
@Frans-Lukas the better way would have been to just mock this dependency of HistoryDao. This is mostly a test of BoxStore and not HistoryDao

Reply:
How do you mean? I don't get what you mean by mocking the dependency of HistoryDao.

Reply:
`boxStore.boxFor()` that shouldn't be here, replace it with a mock. You don't need to create a database for a unit test

Reply:
I think I get how to mock the box for the deleteHistory tests by checking that the box is correctly used with `verify { mockedBox.remove(historyList) }`

But how would I check that e.x.
```
fun saveHistory(historyItem: HistoryItem) {
    box.store.callInTx {
      box
        .query {
          equal(HistoryEntity_.historyUrl, historyItem.historyUrl).and()
            .equal(HistoryEntity_.dateString, historyItem.dateString)
        }
        .remove()
      box.put(HistoryEntity(historyItem))
    }
}
```

works?

Reply:
`every{ box.store } returns mockStore`
You might have to use an argumentcaptor for the Callable passed to `callInTx` and then invoke the callable to continue the tests.
For the query method you might have to use [constructor mocks](https://mockk.io/#constructor-mocks) for the QueryBuilder that gets created.

you should end up with with
```
verify{
mockedQuery.remove()
box.put(HistoryItem)
}
```
and in the process of creating the mocked query the mocks should have been called with the appropriate parameters

```
every {mockedQueryBuilder.equal(HistoryEntity_.historyUrl, historyItem.historyUrl)} returns mockedQueryBuilder
```
and so on